src/iface/interface.rs

```diff
@@ -778,6 +778,33 @@ impl<'a> IpPacket<'a> {
     }
 }
 
+/// Struct only used internally. We want to be able to use `process_udp` for every medium we
```

Comment:
This sounds very very complicated. Why not pass the parsed UdpRepr (src+dst ports) and the UDP payload to `process_udp` instead? This way each caller can do the parsing accordingly. (if there's multiple callers to `process_udp` uncompressed, perhaps having a `process_ip_udp` fn could help avoid duplicating parsing code?).

Reply:
I changed it. But I'm still not happy about it. The ICMP reply for when the destination is unreachable makes everything more difficult.

Reply:
I guess it always has been incorrect for 6LoWPAN.